src/test/java/org/assertj/core/error/ShouldContainExactly_create_Test.java

```diff
@@ -30,6 +30,44 @@ class ShouldContainExactly_create_Test {
 
   private static final ComparatorBasedComparisonStrategy CASE_INSENSITIVE_COMPARISON_STRATEGY = new ComparatorBasedComparisonStrategy(CaseInsensitiveStringComparator.instance);
 
+  @Test
+  void should_display_full_expected_and_actual_sets_when_order_does_not_match() {
+    // GIVEN
+    ErrorMessageFactory factory = shouldContainExactly(list("Yoda", "Han", "Luke", "Anakin"), list("Yoda", "Luke", "Han", "Anakin"),
+                                                       Collections.emptyList(), Collections.emptyList());
+
+    // WHEN
+    final String message = factory.create(new TextDescription("Test"));
+
+    // THEN
+    then(message).isEqualTo(format("[Test] %n"
+                                   + "Expecting actual:%n"
+                                   + "  [\"Yoda\", \"Han\", \"Luke\", \"Anakin\"]%n"
+                                   + "to contain exactly (and in same order):%n"
+                                   + "  [\"Yoda\", \"Luke\", \"Han\", \"Anakin\"]%n"));
```

Comment:
I think what would be more helpful here in addition to this description is to have the list of mismatches, each mismatch showing the index, the actual element value and the expected element value.

On the example above we could add
```
but there were differences at these indexes:
- index 1: actual element "Han" but expected was "Luke"
- index 2: actual element "Luke" but expected was "Han"
```
or
On the example above we could add
```
but there were differences at these indexes:
- element at index 1: expected "Han" but was "Luke"
- element at index 2: expected "Luke" but was "Han"
```

Reply:
I prefer option 2, WDYT?

Reply:
I'm afraid the list of mismatches could be huge (e.g. both lists like 1000 elements and they are in the reverse order), so there should be a hard limit on the number of printed mismatches

Reply:
Option 3:

```diff
@@ -1,4 +1,4 @@
 Yoda
-Han
 Luke
+Han
 Anakin
```

Reply:
> Wouldn't almost all options be big and perhaps unreadable if the lists contain lots of elements?

Well, if the order of elements is completely different, then, I believe, any output that mentions all differences would be too verbose.

That is why I think it might be OK to show only the first (or the first N<=10) differences.
If the number of actual diffs is small, then all of them will be visible. If the number of diffs is big, then the first will be printed, and the user would understand "everything is different".

> without color differences to support the - and + signs it might be difficult to spot the differences

The idea behind + and - is to use unified diff. I moved it to a separate issue https://github.com/assertj/assertj-core/issues/2641

> I'd think the creator of the unit test should consider smaller data sets to prove the functionality is working

Databases support up to ~1000 columns, and I have functional tests that verify if the system is workable with that number of columns. I would still use regular assertions even for collections up to 100'000 elements.
Here's a recent case for a functional test with a collection having 65535 elements: https://github.com/pgjdbc/pgjdbc/pull/2525

Reply:
Let's leave the length error aside (might be tackled in a different issue) and focus the error to produce when the compared iterables have the same number of elements.

I see two kind of errors from here:

1. elements don't match ignoring the order, there might be unexpected / missing elements in actual
2. elements match but the order is not right

For 1. we could show the diff as done in `should_display_full_expected_and_actual_sets_with_missing_and_unexpected_elements` and for 2. show the index mismatches, limiting them for brevity them number (could be configured with Configuration.maxElementsForPrinting which is set to 1000 be default).

Reply:
1000 is too much, I'm happy if we go with the first 10 and if we have more indicate in the error message that we only display the first 10

Reply:
I was playing around with the amount of indexes to display yesterday and the `MAX_ELEMENTS_FOR_PRINTING` is a lot indeed. The first 10 to 50 seems reasonable to me. It could be useful feedback to display something along the lines:
```text
but there were differences at these indexes:
  element at index 0: expected "Luke" but was "Han"
  ...
  element at index 9: expected "Anakin" but was "Leia"
Only showing the first 10 mismatches 
```
WDYT?

Reply:
What output would you suggest for the following call `assertThat(Arrays.asList(10, 1, 2, 3, 4, 5, 6, 7, 8, 9)).containsExactly(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);` ?
I guess displaying all the indices as mismatching won't be that helpful.

Reply:
@Giovds LGTM, maybe we should move `Only showing the first 10 mismatches` before showing the indexes to be sure users will see it (I'm fine with it at the end too, it's just a suggestion)
```
but there were differences at these indexes (only showing the first 10 mismatches):
  element at index 0: expected "Luke" but was "Han"
  ...
  element at index 9: expected "Anakin" but was "Leia"
```
If we have less than 10 mismatches, no need to mention `only showing the first 10 mismatches` 

this is a bit of a contrived example @vlsi, I feel the proposed error message is still helpful enough in this case.

Reply:
I'm also happy to increase to 50 mismatches by default

Reply:
Why do you mean the example is contrieved? If the use-case is to assert elements in a strict order, then it would likely that failure cases would be "almost matching" like I show here. There was almost exact match, except the last element is misplaced